### PR TITLE
Refactoring user profile interfaces and consolidating user representation for both admin and account contexts

### PR DESCRIPTION
**Pull request by pedroigor**

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]

File: server-spi/src/main/java/org/keycloak/userprofile/Attributes.java [hunk 1]
```diff
@@ -19,20 +19,38 @@
...
+ * <li>Managed
+ * <li>Unmanaged
+ * </ul>
+ *
+ * <p>A <i>managed</i> attribute is any attribute defined in the user profile configuration. Therefore, they are known by
+ * the server and can be managed accordingly.
+ *
+ * <p>A <i>unmanaged</i> attributes is any attribute <b>not</b> defined in the user profile configuration. Therefore, the server
+ * does not know about them and they can't leverage capabilities provided by the server. However, they can still be managed by
+ * administrators by setting any of the {@link org.keycloak.representations.userprofile.config.UPConfig.UnmanagedAttributePolicy}.
```
Comment (andymunro) [inline server-spi/src/main/java/org/keycloak/userprofile/Attributes.java:47]:
```suggestion
 * does not know about them and they cannot use capabilities provided by the server. However, they can still be managed by
 * administrators by setting any of the {@link org.keycloak.representations.userprofile.config.UPConfig.UnmanagedAttributePolicy}.
```

**Review by andymunro — CHANGES_REQUESTED** [review]
Looks good. A few changes to suggest @mposolda

**Comment by pedroigor** [comment]
Thanks @andymunro, please let me know if anything else is missing.

**Review by mposolda — APPROVED** [review]

**Comment by mposolda** [comment]
@andymunro Thanks for the docs review!

@pedroigor LGTM, we can adjust later if there are some issues or more things to improve...

**Pull request merged**